### PR TITLE
[Woo POS] Totals calculation app-side issue

### LINKED ISSUE #12496: [Woo POS] Total is wrongly rounded with product prices inclusive in tax
**Describe the bug**

Totals are wrongly rounded with product prices inclusive of tax. This at least partially looks like a POS bug since the card payment modal shows a correct price.

**To Reproduce**

### Settings

I used two Taxable products with Standard rate.

Example steps:

1. Configure store tax settings
2. Open POS
3. Add 2 x $0.01 and 1x $16 products
4. Observe total is $16.03
5. Collect card payment
6. Observe total is $16.02 

**Screenshots**

#### Card Payment Modal shows a different price

image

#### iOS POS vs Android POS vs Web

image
image

**Expected behavior**

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active,
  or [Storefront](https://woocommerce.com/products/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device:
- Android version:
- WooCommerce Android version:

**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
</details>

### PR DESCRIPTION
**Pull request by samiuelson**
Closes: #12496 

### Description

This PR fixes the app-side totals rendering issues, caused by doing local calculations for the values that are available in other fields in the API response.

### Steps to reproduce

Detailed information on how to reproduce the issue, including test store access can be found here: p1726066991535989-slack-C070SJRA8DP

It turns out the backend does its own total values rounding and our local arithmetic calculations were not matching the backend response. The issue is visible e.g. with items containing low prices, sample order item list:
1. 2x $0.02
2. 1x $16.00

See the table with screenshots below. 

Additionally, the test store must have tax rates defined.

### Testing information

Testing should verify that order totals are presented correctly and consistently across iOS and Web. 

### The tests that have been performed

I recreated the scenario from the original issue, using the test store mentioned above. Then confirmed this PR fixes the issue. Additionally, I verified the totals are presented consistently on iOS and Web for the same order. Device: Galaxy Tab S9 FE, Android 14.

### Images/gif

|Before|After|
|---|---|
|Screenshot_20240911_190313|Screenshot_20240911_190030|

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsRepository.kt [hunk 1]
```diff
@@ -43,7 +43,6 @@ class WooPosTotalsRepository @Inject constructor(
                                 quantity = quantity.toFloat(),
                                 total = EMPTY_TOTALS_SUBTOTAL_VALUE,
                                 subtotal = EMPTY_TOTALS_SUBTOTAL_VALUE,
-                                price = EMPTY_TOTALS_SUBTOTAL_VALUE,
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsRepository.kt:46]: It is not related to the fix. I just noticed that in the store management we don't override the price and as per my tests it's redundant.

**Comment by backwardstruck** [comment]
Did you mean to change this back to draft or is it ready for review?

**Comment by samiuelson** [comment]
> Did you mean to change this back to draft or is it ready for review?

@backwardstruck, I realized I need to update tests as well. It's ready for review now!

**Comment by backwardstruck** [comment]
> > Did you mean to change this back to draft or is it ready for review?
> 
> @backwardstruck, I realized I need to update tests as well. It's ready for review now!

Great. Thanks. Working on getting set up to reproduce/test.

**Review by backwardstruck — COMMENTED** [review]
Tested on this branch and `trunk`. Was able to reproduce the issue with the test store and see it fixed here.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsRepository.kt (hunk 1 above)
Comment (backwardstruck) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsRepository.kt:46]: Good catch.

**Review by backwardstruck — APPROVED** [review]
Tested with large and small total amounts. Good change looks good.

**Pull request merged**